AppController/djinn.rb

```diff
@@ -4883,7 +4890,11 @@ def check_stopped_apps()
       if my_node.is_load_balancer?
         stop_xmpp_for_app(app)
         Nginx.remove_app(app)
+
+        # Since the removal of an app from HAProxy can cause a reset of
+        # the drain flags, let's set them again.
         HAProxy.remove_app(app)
+        reset_drain
```

Comment:
Does this ensure that even if requests go through in between the resets, at some point the drain will complete?